pandas/core/arrays/interval.py

```diff
@@ -561,6 +563,106 @@ def from_tuples(
 
         return cls.from_arrays(left, right, closed, copy=False, dtype=dtype)
 
+    _interval_shared_docs["from_strings"] = textwrap.dedent(
+        """
+        Construct from string representations of the left and right bounds.
+
+        Parameters
+        ----------
+        data : array-like (1-dimensional)
+            Strings representing the Interval's to parse.
+        copy : bool, default False
+            Copy the data.
+        dtype : dtype, optional
+            If None, dtype will be inferred.
+
+        Returns
+        -------
+        %(klass)s
+
+        Raises
+        ------
+        ValueError
+            When a string cannot be parsed as an Interval
+            When the dtype of the string cannot be parsed as either float,
+            Timestamp or Timedelta
+
+        See Also
+        --------
+        interval_range : Function to create a fixed frequency IntervalIndex.
+        %(klass)s.from_breaks : Construct an %(klass)s from an array of
+            splits.
+        %(klass)s.from_tuples : Construct an %(klass)s from an
+            array-like of tuples.
+
+        %(examples)s\
+        """
+    )
+
+    @classmethod
+    @Appender(
+        _interval_shared_docs["from_strings"]
+        % {
+            "klass": "IntervalIndex",
+            "examples": textwrap.dedent(
+                """\
+        Examples
+        --------
+        >>> pd.IntervalIndex.from_strings(["(0, 1]", "(1, 2]"])
+        IntervalIndex([(0, 1], (1, 2]],
+                       dtype='interval[int64, right]')
+        """
+            ),
+        }
+    )
+    def from_strings(
+        cls: type[IntervalArrayT],
+        data: Sequence[str],
+    ) -> IntervalArrayT:
+        # These need to be imported here to avoid circular dependencies.
+        from pandas.core.tools.datetimes import to_datetime
+        from pandas.core.tools.timedeltas import to_timedelta
+
+        intervals: list[Interval] = []
+        for string in data:
+
+            # Try to match "(left, right]" where 'left' and 'right' are breaks.
+            breaks_match = re.match(r"\(.*,.*]", string)
```

Comment:
compile this regex

Reply:
you can likely make this support all of the closed types i think (see above)

Reply:
is this regex actually strict enough? e.g. you should require this to match exactly (but whitespace is prob ok)

Reply:
Complied regex in 162045a.

All string representations of the closed types are the same (maybe room for future improvement here?), so from 162045a, the user can specify the closed type as an argument.

Regarding the strictness, could you give an example for when this may pass incorrectly? In the tests, I'm trying different incorrect cases and can't find any undefined behaviour.